### DISCUSSION TITLE
Use spancat with a Knowledge Base

### DISCUSSION THREAD

**Discussion by a1092** [explosion/spaCy#12520] [Help: Coding & Implementations] [answered]
Hello,

Is it possible to use spancat with a KB in order to identify label with an external ID ?

Best regards,

Alex.

**Comment by victorialslocum** [comment, marked as answer]
Hi Alex! In order to assign a KB ID, you’ll need an [entity linking module](https://spacy.io/usage/linguistic-features#entity-linking) within your pipeline. You can use the [EntityLinker](https://spacy.io/api/entitylinker) component within spaCy, for more resources you can check out [this video on Training a custom entity linking model](https://youtu.be/8u57WSXVpmw) or [this talk about entity linking from spaCy IRL](https://youtu.be/PW3RJM8tDGo) or [this example project](https://github.com/explosion/projects/tree/v3/tutorials/nel_emerson).

To add the Entity Linker component to your pipeline with Spancat, you’ll have to save the entities into the `doc.ents` property.  While NER models save directly to `doc.ents`, Spancat automatically stores the output into `doc.spans`, but you should be able to copy the spans over to the `doc.ents` property if the spans are non-overlapping. You can then run the EntityLinker to link the spans to KB IDs.
